components/game/game.model.ts

```diff
@@ -236,6 +250,24 @@ export class Game extends Model {
 		);
 	}
 
+	hasPerms(required?: Perm | Perm[], either?: boolean) {
+		if (!this.perms) {
+			return false;
+		}
+
+		if (!required || this.perms.indexOf('all') !== -1) {
+			return true;
+		}
+
+		required = Array.isArray(required) ? required : [required];
+		const missingPerms = required.filter(perm => this.perms!.indexOf(perm) === -1);
```

Comment:
would be faster to just `for` over it and return immediately when there's a match instead of filtering, then checking length and stuff?